yarp/yarp_compiler.c

```diff
@@ -381,9 +1033,154 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
           }
           return;
       }
+      case YP_NODE_SCOPE_NODE: {
+          yp_scope_node_t *scope_node = (yp_scope_node_t *)node;
+          yp_constant_id_list_t locals = scope_node->locals;
+
+          yp_parameters_node_t *parameters_node = (yp_parameters_node_t *)scope_node->parameters;
+          yp_node_list_t requireds_list = YP_EMPTY_NODE_LIST;
+          yp_node_list_t optionals_list = YP_EMPTY_NODE_LIST;
+
+
+          if (parameters_node) {
+              requireds_list = parameters_node->requireds;
+              optionals_list = parameters_node->optionals;
+          }
+
+          size_t size = locals.size;
+
+          // Index lookup table buffer size is only the number of the locals
+          st_table *index_lookup_table = st_init_numtable();
+
+          VALUE idtmp = 0;
+          rb_ast_id_table_t *tbl = ALLOCV(idtmp, sizeof(rb_ast_id_table_t) + size * sizeof(ID));
```

Comment:
This is just a comment for later, but it looks like we _may_ need to free this? Depending on the size. Other places in the codebase use `ALLOCV_END`. We might need to look into this.